pom.xml

```diff
@@ -190,6 +190,9 @@
             <goals>
               <goal>cpu-count</goal>
             </goals>
+            <configuration>
+              <factor>0.25</factor>
+            </configuration>
```

Comment:
1 fork per core causes a lot of thrashing and really slows things, and also seems to cause flaky tests due to timeouts. This seems more stable.

Reply:
We should also consider GH Action & rate limiting. So, may be it's a good idea to define a general test timeout  through a constant (for parallel ones) and increase it a bit along with this change :)

Reply:
I haven't seen issues in GH, and this change is already big.

I think what you're proposing is a good idea, but it should be done separately to not make this PR even bigger.

Reply:
Agree, let's focus it through a dedicated issue :)